api/src/k8s_client.rs

```diff
@@ -488,11 +514,76 @@ impl K8sClient for HttpK8sClient {
                 phase
             })
             .unwrap_or(PodPhase::Unknown);
+
         Ok(phase)
     }
 
+    async fn get_replicator_container_error(
```

Comment:
In the future we might want to generalize this function for multiple pods but right now I preferred to make it explicit on the replicator container.